### ISSUE TITLE
React Pacakger server - getBoolOptionFromQuery error

### ISSUE THREAD

**Issue by imaffett** [facebook/react-native#218]
Using the starter app, the node server fails.  I'd submit a fix, but can't sign the CLA right now.

```
TypeError: Cannot read property 'dev' of null
    at getBoolOptionFromQuery (/Users/idmaffet/react-native/IanTest/node_modules/react-native/packager/react-packager/src/Server/index.js:276:12)
```

It's an easy fix - just modify the if statement to check if the query object exists

```
function getBoolOptionFromQuery(query, opt, defaultVal) {
   if (!query||(query[opt] == null && defaultVal != null)) {
     return defaultVal;
   }

  return query[opt] === 'true' || query[opt] === '1';
}
```

**Comment by amasad** [comment]
I think your node version is outdated. Can you run `node -v`?
But we should definitely handle that.

**Comment by imaffett** [comment]
v0.11.14

**Comment by amasad** [comment]
yes that version of node has a bug with parsing the url query. Here is the correct behavior http://repl.it/fdz/1

On node v0.11.14 it's null. I'll add the fix anyways. Thanks
```js
console.log(
    "should be an object: ",
    url.parse('/path', true).query
);
```

**Comment by imaffett** [comment]
Thanks for the fast response.

**Comment by lvyile** [comment]
I upgraded node from v0.11.14 to v0.12.1, it works!

**Comment by sosoyososo** [comment]
The same problem on v0.13.0-pre .

**Comment by amasad** [comment]
fixed via 0c83766e3b7d17674954bacf9ccc308d4dd4b2b3
